src/cacheHttpClient.ts

```diff
@@ -82,12 +83,13 @@ function createHttpClient(): HttpClient {
     );
 }
 
-export function getCacheVersion(): string {
+export function getCacheVersion(compressionMethod?: CompressionMethod): string {
     // Add salt to cache version to support breaking changes in cache entry
-    const components = [
-        core.getInput(Inputs.Path, { required: true }),
-        versionSalt
-    ];
+    const components = [core.getInput(Inputs.Path, { required: true })].concat(
+        compressionMethod == CompressionMethod.Zstd
+            ? [compressionMethod, versionSalt]
+            : versionSalt
+    );
```

Comment:
```suggestion
    const components = [core.getInput(Inputs.Path, { required: true }), versionSalt].concat(
        compressionMethod == CompressionMethod.Zstd
            ? [compressionMethod]
            : []
    );
```


This might be more clear if we move `versionSalt` out of the `concat` since it's not dependent on the compression method. (Note this suggested change may need to be `prettier`)